fix the libpcap check so that people can set the cache version when cross-compiling

patch by Shawn Wallace <sjwallace@google.com>

--- a/m4/sf.m4
+++ b/m4/sf.m4
@@ -135,7 +135,7 @@
         echo
         exit 1
     fi
-    AC_MSG_CHECKING([for libpcap version >= $1])
+    AC_CACHE_CHECK([for libpcap version >= $1], [daq_cv_libpcap_version_1x], [
     AC_RUN_IFELSE(
     [AC_LANG_PROGRAM(
     [[
@@ -147,16 +147,13 @@
         if (strcmp(pcap_version, $1) < 0)
             return 1;
     ]])],
-    [libpcap_version_1x="yes"],
-    [libpcap_version_1x="no"])
-    if test "x$libpcap_version_1x" = "xno"; then
-        AC_MSG_RESULT(no)
+    [daq_cv_libpcap_version_1x="yes"],
+    [daq_cv_libpcap_version_1x="no"])])
+    if test "x$daq_cv_libpcap_version_1x" = "xno"; then
         echo
         echo "    ERROR!  Libpcap library version >= $1  not found."
         echo "    Get it from http://www.tcpdump.org"
         echo
         exit 1
-    else
-        AC_MSG_RESULT(yes)
     fi
 ])

